graphql/util.go

```diff
@@ -767,10 +767,21 @@ func ModifyVersionHandler(ctx context.Context, dataConnector data.Connector, pat
 			}
 			if p.IsParent() {
 				for _, childPatchId := range p.Triggers.ChildPatches {
-					err = ModifyVersionHandler(ctx, dataConnector, childPatchId, modifications)
+					p, err := patch.FindOneId(childPatchId)
 					if err != nil {
-						return errors.Wrap(mapHTTPStatusToGqlError(ctx, httpStatus, err), fmt.Sprintf("error modifying child patch '%s'", patchID))
+						return ResourceNotFound.Send(ctx, fmt.Sprintf("error finding child patch %s: %s", childPatchId, err.Error()))
 					}
+					if p == nil {
+						return ResourceNotFound.Send(ctx, fmt.Sprintf("child patch '%s' not found ", childPatchId))
+					}
+					// only modify the child patch if it is finalized
+					if p.Activated {
```

Comment:
more specifically what you're checking here is "does a version exist" which you can check by looking at p.Version instead

Reply:
Is that more fool-proof that checking the Activated field?

Reply:
Theoretically it's the same but just more general; the version definitely exists if the version field is set, whereas I don't remember if Activated can be changed. They probably both work though.

Reply:
checking p.version seems more accurate. I'll change it.